src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,174 @@
+            _localizer[key, arguments];
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+            var tokens = new List<string>();
+
+            for (int i = 0; i < stringValue.Length; i++)
+            {
+                if (stringValue[i] == '{')
+                {
+                    StringBuilder myToken = new StringBuilder();
+                    myToken.Append(stringValue[i]);
+                    i++;
+
+                    while (i < stringValue.Length)
+                    {
+                        if (stringValue[i] != '}')
+                        {
+                            if (stringValue[i] == '{')
+                            {
+                                throw new Exception("Malformed resource string");
+                            }
+                            else
+                            {
+                                myToken.Append(stringValue[i]);
+                                i++;
+                            }
+                        }
+                        else
+                        {
+                            myToken.Append(stringValue[i]);
+                            tokens.Add(myToken.ToString());
+                            break;
+                        }
+                    }
+                }
+            }
+            // This call joins all tokens and formats them based on arguments and splits them again into arguments. The
+            // idea here is to encode the formatted arguments. For example if resource value has datetime format as
+            // argument ex: {1:yyyy} we are formatting the argument to right value before encoding.
+            arguments = string.Format(string.Join(",", tokens), arguments).Split(',');
```

Comment:
Lastly, because this is parsing code, I think more unit tests for all the various escape sequences need to be added.

I might add that there's a somewhat-known `string.Format` "gotcha" that we'll need to account for here: http://blogs.msdn.com/b/brada/archive/2003/12/26/50978.aspx